### ISSUE TITLE
cd rip   is not able to rip the last track

### ISSUE THREAD

**Issue by sp31415t1** [JoeLametta/whipper#203]
I just installed on Debian Stretch.
I tried two cds and I get similar errors : 
   cd info   crashing with AttributeError: 'Namespace' object has no attribute 'offset'
  cd rip    not able to rip the last track.
Is there an issue with software, my install or CDs ?
(message are not really obvious)

full output:

whipper cd info
Checking device /dev/sr0
CDDB disc id: b40ab30d
MusicBrainz disc id HA2E1ss.mlzBsxhtCeMzxuyeqpc-
MusicBrainz lookup URL https://musicbrainz.org/cdtoc/attach?toc=1+13+205622+150+13391+30802+46464+65178+81163+97208+109130+123307+141249+154789+171188+188743&tracks=13&id=HA2E1ss.mlzBsxhtCeMzxuyeqpc-
Disc duration: 00:45:39.626, 13 audio tracks

Matching releases:

Artist  : Stromae
Title   : Racine carrée
Duration: 00:45:36.465
URL     : https://musicbrainz.org/release/de57c1d9-5e65-420f-a896-1332e87d4c09
Release : de57c1d9-5e65-420f-a896-1332e87d4c09
Type    : Album
Barcode : 602537479870
Cat no  : 37479870

Traceback (most recent call last):
  File "/home/sp31415/apps/whipper/master-20171029-post_0.5.1/bin/whipper", line 11, in <module>
    load_entry_point('whipper', 'console_scripts', 'whipper')()
  File "/home/sp31415/apps/whipper/master-20171029-post_0.5.1/whipper/whipper/command/main.py", line 32, in main
    ret = cmd.do()
  File "/home/sp31415/apps/whipper/master-20171029-post_0.5.1/whipper/whipper/command/basecommand.py", line 124, in do
    return self.cmd.do()
  File "/home/sp31415/apps/whipper/master-20171029-post_0.5.1/whipper/whipper/command/basecommand.py", line 124, in do
    return self.cmd.do()
  File "/home/sp31415/apps/whipper/master-20171029-post_0.5.1/whipper/whipper/command/cd.py", line 150, in do
    self.device, self.options.offset)
AttributeError: 'Namespace' object has no attribute 'offset'

$ whipper cd rip
Using configured read offset 667
Checking device /dev/sr0
CDDB disc id: b40ab30d
MusicBrainz disc id HA2E1ss.mlzBsxhtCeMzxuyeqpc-
MusicBrainz lookup URL https://musicbrainz.org/cdtoc/attach?toc=1+13+205622+150+13391+30802+46464+65178+81163+97208+109130+123307+141249+154789+171188+188743&tracks=13&id=HA2E1ss.mlzBsxhtCeMzxuyeqpc-
Disc duration: 00:45:39.626, 13 audio tracks

Matching releases:

Artist  : Stromae
Title   : Racine carrée
Duration: 00:45:36.465
URL     : https://musicbrainz.org/release/de57c1d9-5e65-420f-a896-1332e87d4c09
Release : de57c1d9-5e65-420f-a896-1332e87d4c09
Type    : Album
Barcode : 602537479870
Cat no  : 37479870

creating output directory ./album/Stromae - Racine carrée
Ripping track 1 of 13: 01. Stromae - Ta fête.flac
CRCs match for track 1                    
Peak level: 99.04% 
Rip quality: 100.00%
Ripping track 2 of 13: 02. Stromae - Papaoutai.flac
CRCs match for track 2                    
Peak level: 99.14% 
Rip quality: 100.00%
Ripping track 3 of 13: 03. Stromae - Bâtard.flac
CRCs match for track 3                    
Peak level: 99.15% 
Rip quality: 100.00%
Ripping track 4 of 13: 04. Stromae - Ave Cesaria.flac
CRCs match for track 4                    
Peak level: 99.44% 
Rip quality: 100.00%
Ripping track 5 of 13: 05. Stromae - Tous les mêmes.flac
CRCs match for track 5                    
Peak level: 98.96% 
Rip quality: 100.00%
Ripping track 6 of 13: 06. Stromae - Formidable.flac
CRCs match for track 6                    
Peak level: 99.11% 
Rip quality: 100.00%
Ripping track 7 of 13: 07. Stromae - Moules frites.flac
CRCs match for track 7                    
Peak level: 99.50% 
Rip quality: 100.00%
Ripping track 8 of 13: 08. Stromae - Carmen.flac
CRCs match for track 8                    
Peak level: 99.73% 
Rip quality: 100.00%
Ripping track 9 of 13: 09. Stromae - Humain à l'eau.flac
CRCs match for track 9                    
Peak level: 99.24% 
Rip quality: 100.00%
Ripping track 10 of 13: 10. Stromae - Quand c'est _.flac
CRCs match for track 10                    
Peak level: 98.99% 
Rip quality: 100.00%
Ripping track 11 of 13: 11. Stromae - Sommeil.flac
CRCs match for track 11                    
Peak level: 99.02% 
Rip quality: 100.00%
Ripping track 12 of 13: 12. Stromae - Merci.flac
CRCs match for track 12                    
Peak level: 99.10% 
Rip quality: 100.00%
Ripping track 13 of 13: 13. Stromae avec Maître Gims & OrelSan - Avf.flac
WARNING:whipper.program.cdparanoia:file size 39616556 did not match expected size 39699452
WARNING:whipper.program.cdparanoia:non-integral amount of frames difference
Ripping track 13 of 13 (try 2): 13. Stromae avec Maître Gims & OrelSan - Avf.flac
WARNING:whipper.program.cdparanoia:file size 39616556 did not match expected size 39699452
WARNING:whipper.program.cdparanoia:non-integral amount of frames difference
Ripping track 13 of 13 (try 3): 13. Stromae avec Maître Gims & OrelSan - Avf.flac
WARNING:whipper.program.cdparanoia:file size 39616556 did not match expected size 39699452
WARNING:whipper.program.cdparanoia:non-integral amount of frames difference
Ripping track 13 of 13 (try 4): 13. Stromae avec Maître Gims & OrelSan - Avf.flac
WARNING:whipper.program.cdparanoia:file size 39616556 did not match expected size 39699452
WARNING:whipper.program.cdparanoia:non-integral amount of frames difference
Ripping track 13 of 13 (try 5): 13. Stromae avec Maître Gims & OrelSan - Avf.flac
WARNING:whipper.program.cdparanoia:file size 39616556 did not match expected size 39699452
WARNING:whipper.program.cdparanoia:non-integral amount of frames difference
CRITICAL:whipper.command.cd:Giving up on track 13 after 5 times
track can't be ripped. Rip attempts number is equal to 'MAX_TRIES'

whipper cd info 
Checking device /dev/sr0
CDDB disc id: c40c260e
MusicBrainz disc id ybLIBaJFdm7eRHSikGQojQF7gzQ-
MusicBrainz lookup URL https://musicbrainz.org/cdtoc/attach?toc=1+14+233426+150+16570+30952+45227+63471+74731+89011+104228+123046+147633+165759+179801+196851+214153&tracks=14&id=ybLIBaJFdm7eRHSikGQojQF7gzQ-
Disc duration: 00:51:50.346, 14 audio tracks

Matching releases:

Artist  : Nina Simone
Title   : The Essential Collection (Disc 1 of 2)
Duration: 00:51:48.931
URL     : https://musicbrainz.org/release/f2ef83ef-c7d6-4f8f-9a89-99d71156394b
Release : f2ef83ef-c7d6-4f8f-9a89-99d71156394b
Type    : Compilation
Cat no  : METRDCD513

Traceback (most recent call last):
  File "/home/sp31415/apps/whipper/master-20171029-post_0.5.1/bin/whipper", line 11, in <module>
    load_entry_point('whipper', 'console_scripts', 'whipper')()
  File "/home/sp31415/apps/whipper/master-20171029-post_0.5.1/whipper/whipper/command/main.py", line 32, in main
    ret = cmd.do()
  File "/home/sp31415/apps/whipper/master-20171029-post_0.5.1/whipper/whipper/command/basecommand.py", line 124, in do
    return self.cmd.do()
  File "/home/sp31415/apps/whipper/master-20171029-post_0.5.1/whipper/whipper/command/basecommand.py", line 124, in do
    return self.cmd.do()
  File "/home/sp31415/apps/whipper/master-20171029-post_0.5.1/whipper/whipper/command/cd.py", line 150, in do
    self.device, self.options.offset)
AttributeError: 'Namespace' object has no attribute 'offset'

whipper cd rip
Using configured read offset 667
Checking device /dev/sr0
CDDB disc id: c40c260e
MusicBrainz disc id ybLIBaJFdm7eRHSikGQojQF7gzQ-
MusicBrainz lookup URL https://musicbrainz.org/cdtoc/attach?toc=1+14+233426+150+16570+30952+45227+63471+74731+89011+104228+123046+147633+165759+179801+196851+214153&tracks=14&id=ybLIBaJFdm7eRHSikGQojQF7gzQ-
Disc duration: 00:51:50.346, 14 audio tracks

Matching releases:

Artist  : Nina Simone
Title   : The Essential Collection (Disc 1 of 2)
Duration: 00:51:48.931
URL     : https://musicbrainz.org/release/f2ef83ef-c7d6-4f8f-9a89-99d71156394b
Release : f2ef83ef-c7d6-4f8f-9a89-99d71156394b
Type    : Compilation
Cat no  : METRDCD513

creating output directory ./compilation/Nina Simone - The Essential Collection (Disc 1 of 2)
Ripping track 1 of 14: 01. Nina Simone - My Baby Just Cares for Me.flac
CRCs match for track 1                    
Peak level: 86.42% 
Rip quality: 100.00%
Ripping track 2 of 14: 02. Nina Simone - Don't Smoke in Bed.flac
CRCs match for track 2                    
Peak level: 89.02% 
Rip quality: 100.00%
Ripping track 3 of 14: 03. Nina Simone - He's Got the Whole World in His Hands.flac
CRCs match for track 3                    
Peak level: 88.94% 
Rip quality: 100.00%
Ripping track 4 of 14: 04. Nina Simone - Mood Indigo.flac
CRCs match for track 4                    
Peak level: 88.92% 
Rip quality: 100.00%
Ripping track 5 of 14: 05. Nina Simone - He Needs Me.flac
CRCs match for track 5                    
Peak level: 88.91% 
Rip quality: 100.00%
Ripping track 6 of 14: 06. Nina Simone - African Mailman.flac
CRCs match for track 6                    
Peak level: 88.94% 
Rip quality: 100.00%
Ripping track 7 of 14: 07. Nina Simone - Love Me or Leave Me.flac
CRCs match for track 7                    
Peak level: 88.17% 
Rip quality: 100.00%
Ripping track 8 of 14: 08. Nina Simone - I Loves You Porgy.flac
CRCs match for track 8                    
Peak level: 82.13% 
Rip quality: 100.00%
Ripping track 9 of 14: 09. Nina Simone - Good Bait.flac
CRCs match for track 9                    
Peak level: 87.35% 
Rip quality: 100.00%
Ripping track 10 of 14: 10. Nina Simone - For All We Know.flac
CRCs match for track 10                    
Peak level: 88.28% 
Rip quality: 100.00%
Ripping track 11 of 14: 11. Nina Simone - Central Park Blues.flac
CRCs match for track 11                    
Peak level: 89.27% 
Rip quality: 100.00%
Ripping track 12 of 14: 12. Nina Simone - You'll Never Walk Alone.flac
CRCs match for track 12                    
Peak level: 87.61% 
Rip quality: 100.00%
Ripping track 13 of 14: 13. Nina Simone - Plain Gold Ring.flac
CRCs match for track 13                    
Peak level: 84.94% 
Rip quality: 100.00%
Ripping track 14 of 14: 14. Nina Simone - Little Girl Blue.flac
WARNING:whipper.program.cdparanoia:file size 45252652 did not match expected size 45330140
WARNING:whipper.program.cdparanoia:non-integral amount of frames difference
Ripping track 14 of 14 (try 2): 14. Nina Simone - Little Girl Blue.flac
WARNING:whipper.program.cdparanoia:file size 45252652 did not match expected size 45330140
WARNING:whipper.program.cdparanoia:non-integral amount of frames difference
Ripping track 14 of 14 (try 3): 14. Nina Simone - Little Girl Blue.flac
WARNING:whipper.program.cdparanoia:file size 45252652 did not match expected size 45330140
WARNING:whipper.program.cdparanoia:non-integral amount of frames difference
Ripping track 14 of 14 (try 4): 14. Nina Simone - Little Girl Blue.flac
WARNING:whipper.program.cdparanoia:file size 45252652 did not match expected size 45330140
WARNING:whipper.program.cdparanoia:non-integral amount of frames difference
Ripping track 14 of 14 (try 5): 14. Nina Simone - Little Girl Blue.flac
WARNING:whipper.program.cdparanoia:file size 45252652 did not match expected size 45330140
WARNING:whipper.program.cdparanoia:non-integral amount of frames difference
CRITICAL:whipper.command.cd:Giving up on track 14 after 5 times
track can't be ripped. Rip attempts number is equal to 'MAX_TRIES'

**Comment by sp31415t1** [comment]
with debug:
WARNING:whipper.program.cdparanoia:file size 39616556 did not match expected size 39699452
WARNING:whipper.program.cdparanoia:non-integral amount of frames difference
DEBUG:whipper.program.cdparanoia:getTrackQuality: frames 16879, reads 33143
DEBUG:whipper.program.cdparanoia:stop: exception FileSizeError(u'/tmp/tmp49sElE.whipper.wav', 'File size 39616556 did not match expected size 39699452')
DEBUG:whipper.command.cd:Got exception TaskException(FileSizeError(u'/tmp/tmp49sElE.whipper.wav', 'File size 39616556 did not match expected size 39699452'), 'exception %(exc)s at %(filename)s:%(line)s: ') on try 1

maybe a warning can be considered as sufficient (instead of failing).
Is it possible to have an option to bypass ?
Regards
Serge

**Comment by pyropingvin** [comment]
I have the same problem


```
$ whipper cd rip
Using configured read offset 667
Checking device /dev/sr0
CDDB disc id: 8611d80b
MusicBrainz disc id mVhYYWxcoktBfUPLUeC_mMBS9gI-
MusicBrainz lookup URL https://musicbrainz.org/cdtoc/attach?toc=1+11+342757+150+61287+80835+99537+123225+154197+185397+206340+226607+275287+310707&tracks=11&id=mVhYYWxcoktBfUPLUeC_mMBS9gI-
Disc duration: 01:16:08.093, 11 audio tracks

Matching releases:

Artist  : Pink Floyd
Title   : Pulse (Disc 1 of 2)
Duration: 01:16:07.397
URL     : https://musicbrainz.org/release/242f906b-93c9-3409-ae89-5084f39d0eaf
Release : 242f906b-93c9-3409-ae89-5084f39d0eaf
Type    : Live
Cat no  : 7243 8 32701 2 5

Artist  : Pink Floyd
Title   : Pulse (Disc 1 of 2)
Duration: 01:16:07.397
URL     : https://musicbrainz.org/release/24ce2ec1-7b23-32f8-a92c-c576e054159b
Release : 24ce2ec1-7b23-32f8-a92c-c576e054159b
Type    : Live
Barcode : 724383270026
Cat no  : 7243 8 32700 2 6

Artist  : Pink Floyd
Title   : Pulse (Disc 1 of 2)
Duration: 01:16:07.397
URL     : https://musicbrainz.org/release/64c0bf20-62e3-4b61-90d5-26a40e1f0c78
Release : 64c0bf20-62e3-4b61-90d5-26a40e1f0c78
Type    : Live
Barcode : 724383270026
Cat no  : CDEMD 1078

Artist  : Pink Floyd
Title   : Pulse (Disc 1 of 2)
Duration: 01:16:07.397
URL     : https://musicbrainz.org/release/76031424-5fd4-3948-a1f0-60eccc7c6d6e
Release : 76031424-5fd4-3948-a1f0-60eccc7c6d6e
Type    : Live
Cat no  : C2K 67065

Artist  : Pink Floyd
Title   : Pulse (Disc 1 of 2)
Duration: 01:16:07.397
URL     : https://musicbrainz.org/release/f7725225-afbf-30eb-9442-a0a28ae82f5a
Release : f7725225-afbf-30eb-9442-a0a28ae82f5a
Type    : Live
Barcode : 724349143623
Cat no  : 7243 4 91436 2 3

creating output directory /home/zivan/Music/RIP/live/Pink Floyd - Pulse (Disc 1 of 2)
Ripping track 1 of 11: 01. Pink Floyd - Shine On You Crazy Diamond.flac
CRCs match for track 1                    
Peak level: 100.00% 
Rip quality: 100.00%
Ripping track 2 of 11: 02. Pink Floyd - Astronomy Domine.flac
CRCs match for track 2                    
Peak level: 93.27% 
Rip quality: 100.00%
Ripping track 3 of 11: 03. Pink Floyd - What Do You Want From Me_.flac
CRCs match for track 3                    
Peak level: 94.99% 
Rip quality: 100.00%
Ripping track 4 of 11: 04. Pink Floyd - Learning to Fly.flac
CRCs match for track 4                    
Peak level: 100.00% 
Rip quality: 100.00%
Ripping track 5 of 11: 05. Pink Floyd - Keep Talking.flac
CRCs match for track 5                    
Peak level: 100.00% 
Rip quality: 100.00%
Ripping track 6 of 11: 06. Pink Floyd - Coming Back to Life.flac
CRCs match for track 6                    
Peak level: 86.88% 
Rip quality: 100.00%
Ripping track 7 of 11: 07. Pink Floyd - Hey You.flac
CRCs match for track 7                    
Peak level: 82.23% 
Rip quality: 100.00%
Ripping track 8 of 11: 08. Pink Floyd - A Great Day for Freedom.flac
CRCs match for track 8                    
Peak level: 94.49% 
Rip quality: 100.00%
Ripping track 9 of 11: 09. Pink Floyd - Sorrow.flac
CRCs match for track 9                    
Peak level: 100.00% 
Rip quality: 100.00%
Ripping track 10 of 11: 10. Pink Floyd - High Hopes.flac
CRCs match for track 10                    
Peak level: 90.84% 
Rip quality: 100.00%
Ripping track 11 of 11: 11. Pink Floyd - Another Brick in the Wall, Part 2.flac
WARNING:whipper.program.cdparanoia:file size 75300908 did not match expected size 75381644
WARNING:whipper.program.cdparanoia:non-integral amount of frames difference
Ripping track 11 of 11 (try 2): 11. Pink Floyd - Another Brick in the Wall, Part 2.flac
WARNING:whipper.program.cdparanoia:file size 75300908 did not match expected size 75381644
WARNING:whipper.program.cdparanoia:non-integral amount of frames difference
Ripping track 11 of 11 (try 3): 11. Pink Floyd - Another Brick in the Wall, Part 2.flac
WARNING:whipper.program.cdparanoia:file size 75300908 did not match expected size 75381644
WARNING:whipper.program.cdparanoia:non-integral amount of frames difference
Ripping track 11 of 11 (try 4): 11. Pink Floyd - Another Brick in the Wall, Part 2.flac
WARNING:whipper.program.cdparanoia:file size 75300908 did not match expected size 75381644
WARNING:whipper.program.cdparanoia:non-integral amount of frames difference
Ripping track 11 of 11 (try 5): 11. Pink Floyd - Another Brick in the Wall, Part 2.flac
WARNING:whipper.program.cdparanoia:file size 75300908 did not match expected size 75381644
WARNING:whipper.program.cdparanoia:non-integral amount of frames difference
CRITICAL:whipper.command.cd:Giving up on track 11 after 5 times
track can't be ripped. Rip attempts number is equal to 'MAX_TRIES'
$
```

**Comment by Drakenson** [comment]
Same problem here on Debian Testing. Seems like its missing some bytes at the end of the CD?

```
robin@Asuka:~$ whipper cd rip
Using configured read offset 667
Checking device /dev/sr0
CDDB disc id: c70dcb0f
MusicBrainz disc id 9LMcyKm14tJmS0kiWK6fwJDmASQ-
MusicBrainz lookup URL https://musicbrainz.org/cdtoc/attach?toc=1+15+264978+150+17626+34239+50868+66410+88671+104704+122561+142341+159422+177000+195827+210024+232545+249527&tracks=15&id=9LMcyKm14tJmS0kiWK6fwJDmASQ-
Disc duration: 00:58:51.040, 15 audio tracks

Matching releases:

Artist  : Feuerschwanz
Title   : Auf's Leben!
Duration: 00:58:51.034
URL     : https://musicbrainz.org/release/1aaf5d1f-54d9-4d36-8863-6109cdabaa17
Release : 1aaf5d1f-54d9-4d36-8863-6109cdabaa17
Type    : Album
Barcode : 4260240783927

creating output directory ./album/Feuerschwanz - Auf's Leben!
Ripping track 1 of 15: 01. Feuerschwanz - Auf's Leben.flac
CRCs match for track 1                    
Peak level: 96.61% 
Rip quality: 100.00%
Ripping track 2 of 15: 02. Feuerschwanz - Herz im Sturm.flac
CRCs match for track 2                    
Peak level: 96.61% 
Rip quality: 100.00%
Ripping track 3 of 15: 03. Feuerschwanz - Mann aus Metall.flac
CRCs match for track 3                    
Peak level: 96.61% 
Rip quality: 100.00%
Ripping track 4 of 15: 04. Feuerschwanz - Zuckerbrot und Peitsche.flac
CRCs match for track 4                    
Peak level: 96.61% 
Rip quality: 100.00%
Ripping track 5 of 15: 05. Feuerschwanz - Hans.flac
CRCs match for track 5                    
Peak level: 96.61% 
Rip quality: 100.00%
Ripping track 6 of 15: 06. Feuerschwanz - Blöde Frage, Saufgelage.flac
CRCs match for track 6                    
Peak level: 96.61% 
Rip quality: 100.00%
Ripping track 7 of 15: 07. Feuerschwanz - Seemannsliebe.flac
CRCs match for track 7                    
Peak level: 96.61% 
Rip quality: 100.00%
Ripping track 8 of 15: 08. Feuerschwanz - Auf Wiederseh'n.flac
CRCs match for track 8                    
Peak level: 96.61% 
Rip quality: 100.00%
Ripping track 9 of 15: 09. Feuerschwanz - Träumer und Tor.flac
CRCs match for track 9                    
Peak level: 96.61% 
Rip quality: 100.00%
Ripping track 10 of 15: 10. Feuerschwanz - Sündenfrei (Zum Sonderpreis).flac
CRCs match for track 10                    
Peak level: 98.30% 
Rip quality: 100.00%
Ripping track 11 of 15: 11. Feuerschwanz - Ohrwurm.flac
CRCs match for track 11                    
Peak level: 96.61% 
Rip quality: 100.00%
Ripping track 12 of 15: 12. Feuerschwanz - Der Druide.flac
CRCs match for track 12                    
Peak level: 96.61% 
Rip quality: 100.00%
Ripping track 13 of 15: 13. Feuerschwanz - Frisch Gezapft.flac
CRCs match for track 13                    
Peak level: 96.67% 
Rip quality: 100.00%
Ripping track 14 of 15: 14. Feuerschwanz - In Vino Veritas.flac
CRCs match for track 14                    
Peak level: 96.61% 
Rip quality: 100.00%
Ripping track 15 of 15: 15. Feuerschwanz - Auf's Leben (Nach der Feier).flac
WARNING:whipper.program.cdparanoia:file size 36274220 did not match expected size 36340796
WARNING:whipper.program.cdparanoia:non-integral amount of frames difference
Ripping track 15 of 15 (try 2): 15. Feuerschwanz - Auf's Leben (Nach der Feier).flac
WARNING:whipper.program.cdparanoia:file size 36274220 did not match expected size 36340796
WARNING:whipper.program.cdparanoia:non-integral amount of frames difference
Ripping track 15 of 15 (try 3): 15. Feuerschwanz - Auf's Leben (Nach der Feier).flac
WARNING:whipper.program.cdparanoia:file size 36274220 did not match expected size 36340796
WARNING:whipper.program.cdparanoia:non-integral amount of frames difference
Ripping track 15 of 15 (try 4): 15. Feuerschwanz - Auf's Leben (Nach der Feier).flac
WARNING:whipper.program.cdparanoia:file size 36274220 did not match expected size 36340796
WARNING:whipper.program.cdparanoia:non-integral amount of frames difference
Ripping track 15 of 15 (try 5): 15. Feuerschwanz - Auf's Leben (Nach der Feier).flac
WARNING:whipper.program.cdparanoia:file size 36274220 did not match expected size 36340796
WARNING:whipper.program.cdparanoia:non-integral amount of frames difference
CRITICAL:whipper.command.cd:Giving up on track 15 after 5 times
track can't be ripped. Rip attempts number is equal to 'MAX_TRIES'
```

**Comment by sp31415t1** [comment]
probably related to https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=870101

**Comment by JoeLametta** [comment]
I think this one should be already solved if using a whipper >= `v0.6.0` together with libcdio-paranoia `10.2+0.94+2`.
If using whipper < `v0.6.0` (or morituri) a patched cdparanoia package is needed ([link to the patch](http://lists.xiph.org/pipermail/paranoia-dev/2013-July/000277.html)).

If I'm wrong, please reopen the issue.

**Comment by bjornfor** [comment]
I have the exact same issue as OP. I'm running whipper-0.7.0 from [nix](https://nixos.org/nix/), which includes libcdio-paranoia-10.2+0.94+2. I also tested whipper-0.7.2, but no change.

(I also get crash on "whipper cd info", but I see there is a [bug](https://github.com/whipper-team/whipper/issues/230) on that already.)

whipper calculated drive read offset 667, which appears to be in the problem range mentioned in [this patch](http://lists.xiph.org/pipermail/paranoia-dev/2013-July/000277.html) (which the parent comment linked to).

```
$ whipper drive list
drive: /dev/cdrom, vendor: HL-DT-ST, model: DVDRAM GSA-H62N , release: CL00
       Configured read offset: 667
       Can defeat audio cache: True
```

**Comment by bjornfor** [comment]
> I'm running whipper-0.7.0 from nix, which includes libcdio-paranoia-10.2+0.94+2.

That's not correct. In nix, whipper is currently built with cdparanoia, not libcdio-paranoia. I'll switch it, test and report back.

**Comment by bjornfor** [comment]
After switching from cdparanoia-III-10.2 to libcdio-paranoia-10.2+0.94+2, I'm getting different error:

```
$ whipper cd rip
...
Ripping track 1 of 11: 01. U2 - Gloria.flac
WARNING:whipper.program.cdparanoia:file size 0 did not match expected size 44452844
WARNING:whipper.program.cdparanoia:non-integral amount of frames difference
[...]
CRITICAL:whipper.command.cd:Giving up on track 1 after 5 times
track can't be ripped. Rip attempts number is equal to 'MAX_TRIES'
```

So instead of failing on the last track, it fails on the first, with size 0 error instead of very-close-to-expected-size error it did before (then only on the last track).

Looks like there is a bug for that: https://github.com/whipper-team/whipper/issues/234

**Comment by JoeLametta** [comment]
> Looks like there is a bug for that: #234

Correct: unfortunately it's an upstream issue. Let's leave this ticket closed and keep discussing in #234.
